core/src/main/java/jenkins/install/InstallUtil.java

```diff
@@ -93,10 +93,6 @@ public T get() {
      * Proceed to the state following the provided one
      */
     public static void proceedToNextStateFrom(InstallState prior) {
-        InstallState current = Jenkins.getInstance().getInstallState();
-        if (!current.equals(prior)) {
-            if (Main.isDevelopmentMode) LOGGER.warning("Transitioning state from: " + prior + ", but current is: " + current);
```

Comment:
IMHO it's useful not only in the Development mode. Deletion as a bugfix seems to be a strange approach